src/utils/request.tsx

```diff
@@ -0,0 +1,12 @@
+import axios from "axios";
+
+export const userRequest = axios.create({
+  baseURL: "https://upskilling-egypt.com:3000/api/v0",
+  headers: {
```

Comment:
We should use interceptors for the headers. If the local storage is initially empty, this code will run initially and the token will be null. So all APIs requiring a token will not work. Interceptors work in run time, when we call any API it will be able to get the last value in the localstorage.